File: python/insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def insert(node: Optional[TreeNode], val: int):
        if val < node.val:
            if node.left == None:
                node.left = TreeNode(val=val)
            else:
                Solution.insert(node.left, val)
        else:
            if node.right == None:
                node.right = TreeNode(val=val)
            else:
                Solution.insert(node.right, val)


    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if root == None:
            root = TreeNode(val=val)
        else:
            Solution.insert(root, val)
        return root

File: go/insert-into-a-binary-search-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 func insert(node *TreeNode, val int) {
    if val < node.Val {
        if node.Left == nil {
            node.Left = &TreeNode{Val: val}
        } else {
            insert(node.Left, val)
        }
    } else {
        if node.Right == nil {
            node.Right = &TreeNode{Val: val}
        } else {
            insert(node.Right, val)
        }
    }
}

func insertIntoBST(root *TreeNode, val int) *TreeNode {
    if root == nil {
        root = &TreeNode{Val: val}
    } else {
        insert(root, val)
    }
    return root
}
